mono/metadata/console-unix.c

```diff
@@ -240,9 +241,12 @@ do_console_cancel_event (void)
 		g_assert (cancel_handler_field);
 	}
 
-	vtable = mono_class_vtable_full (domain, klass, FALSE);
-	if (vtable == NULL)
+	vtable = mono_class_vtable_full (domain, klass, &error);
+	if (vtable == NULL) {
+		mono_error_cleanup (&error);
 		return;
+	}
+	mono_error_assert_ok (&error);
```

Comment:
would there be a case where it the error is set, but it doesn't return NULL?
you could also change the above check by `if (!vtable || !is_ok (&error))`